### PR TITLE
Docs Pattern matching. Close: #1500

### PR DESCRIPTION
**Pull request by denysdovhan**
New section added — **Pattern matching**. Links to **anymatch** module has been replaced by links to the **Pattern matching** section.

---

## Pattern matching

Internally, Brunch uses [anymatch](https://github.com/es128/anymatch) for pattern matching — it's a module to match strings against configurable strings, globs, regular expressions or functions. If you want to define specific file pattern, you can use:

* **String** — path to your source files to be directly matched. You can use [globs](https://en.m.wikipedia.org/wiki/Glob_(programming)), so these strings may contain [wildcard characters](https://en.m.wikipedia.org/wiki/Wildcard_character) (`*`, `?`, etc). Example:

  ```javascript
  joinTo: {
    'app.css': 'path/to/*.css' # matches all CSS files
  }
  ```
* **Regular expression** — feel free to define a pattern using regular expression if you used to. Example:

  ```javascript
  joinTo: {
    'app.js': /.js$/ # matches all JavaScript files
  }
  ```
* **Function** — if you need especially specific pattern, you can define it using function that takes the string as an argument and returns a truthy or falsy value (like `Array.filter` function do). Example:

  ```javascript
  joinTo: {
    # matches all JavaScript files with filenames longer that 3 characters
    'app.js': str => str.indexOf('.js') !== -1 && str.length > 3
  }
  ```
* **Array** — an array of any number and mix of the above types. Example:

  ```javascript
  joinTo: {
    'app.js': [
      'path/to/specific/file.js',       # include specific file
      'any/**/*.js',                    # all files with .js extension
      /.es6.js$/,                       # all files with .es6.js extension
      str => str.indexOf('tmp') === -1  # contains `tmp` substring
    ]
  }
  ```

---

cc/ @shvaikalesh @mryarbles

### CODE REVIEW

File: source/app/assets/_docs/config.md [hunk 1]
```diff
@@ -38,6 +38,45 @@ module.exports = {
...
 
+## Pattern matching
+
+Internally, Brunch uses [anymatch](https://github.com/es128/anymatch) for pattern matching — it's a module to match strings against configurable strings, globs, regular expressions or functions. If you want to define specific file pattern, you can use:
+
+* **String** — path to your source files to be directly matched. You can use [globs](https://en.m.wikipedia.org/wiki/Glob_(programming)), so these strings may contain [wildcard characters](https://en.m.wikipedia.org/wiki/Wildcard_character) (`*`, `?`, etc). Example:
+
+  ```javascript
+  joinTo: {
+    'app.css': 'path/to/*.css' # matches all CSS files
```
Comment (shvaikalesh) [inline source/app/assets/_docs/config.md]: `//`

File: source/app/assets/_docs/config.md [hunk 2]
```diff
@@ -38,6 +38,45 @@ module.exports = {
...
+  joinTo: {
+    'app.js': /.js$/ # matches all JavaScript files
+  }
+  ```
+* **Function** — if you need especially specific pattern, you can define it using function that takes the string as an argument and returns a truthy or falsy value (like `Array.filter` function do). Example:
+
+  ```javascript
+  joinTo: {
+    # matches all JavaScript files with filenames longer that 3 characters
+    'app.js': str => str.indexOf('.js') !== -1 && str.length > 3
```
Comment (shvaikalesh) [inline source/app/assets/_docs/config.md]: `.includes`

File: source/app/assets/_docs/config.md [hunk 3]
```diff
@@ -38,6 +38,45 @@ module.exports = {
...
+  ```javascript
+  joinTo: {
+    'app.css': 'path/to/*.css' # matches all CSS files
+  }
+  ```
+* **Regular expression** — feel free to define a pattern using regular expression if you used to. Example:
+
+  ```javascript
+  joinTo: {
+    'app.js': /.js$/ # matches all JavaScript files
```
Comment (shvaikalesh) [inline source/app/assets/_docs/config.md]: `//`

File: source/app/assets/_docs/config.md [hunk 4]
```diff
@@ -38,6 +38,45 @@ module.exports = {
...
+  ```
+* **Array** — an array of any number and mix of the above types. Example:
+
+  ```javascript
+  joinTo: {
+    'app.js': [
+      'path/to/specific/file.js',       # include specific file
+      'any/**/*.js',                    # all files with .js extension
+      /.es6.js$/,                       # all files with .es6.js extension
+      str => str.indexOf('tmp') === -1  # contains `tmp` substring
```
Comment (shvaikalesh) [inline source/app/assets/_docs/config.md]: `.includes`, `//`

File: source/app/assets/_docs/config.md [hunk 5]
```diff
@@ -38,6 +38,45 @@ module.exports = {
...
+  }
+  ```
+* **Array** — an array of any number and mix of the above types. Example:
+
+  ```javascript
+  joinTo: {
+    'app.js': [
+      'path/to/specific/file.js',       # include specific file
+      'any/**/*.js',                    # all files with .js extension
+      /.es6.js$/,                       # all files with .es6.js extension
```
Comment (shvaikalesh) [inline source/app/assets/_docs/config.md]: `/\.es6\.js$/`

File: source/app/assets/_docs/config.md [hunk 6]
```diff
@@ -38,6 +38,45 @@ module.exports = {
 Each Brunch config is an executable script,
 so you can also execute arbitrary JS and import Node.js modules there.
 
+## Pattern matching
+
+Internally, Brunch uses [anymatch](https://github.com/es128/anymatch) for pattern matching — it's a module to match strings against configurable strings, globs, regular expressions or functions. If you want to define specific file pattern, you can use:
+
+* **String** — path to your source files to be directly matched. You can use [globs](https://en.m.wikipedia.org/wiki/Glob_(programming)), so these strings may contain [wildcard characters](https://en.m.wikipedia.org/wiki/Wildcard_character) (`*`, `?`, etc). Example:
```
Comment (denysdovhan) [inline source/app/assets/_docs/config.md]: nice catch

**Comment by denysdovhan** [comment]
Done. Please, check.

**Pull request merged**